scipy/stats/tests/test_stats.py

```diff
@@ -5645,6 +5645,11 @@ def check_equal_pmean(array_like, exp, desired, axis=None, dtype=None,
 
 
 class TestHarMean:
+    def test_0(self):
+        a = [1, 0, 2]
```

Comment:
This is fine. I'm curious what happens if there is a negative zero, though? There is a different representation of negative zero and I'm not sure what the reciprocal of it is. If the reciprocal is `-np.inf`, and if we also have a regular zero, it seems we could end up with an indeterminate result. I'll check when I get home.
(You know, since we're splitting hairs about this edge case already!)

Reply:
Just tested with -0 and it gives 0. So I think this is good.

Reply:
If we choose to define harmonic mean the natural way even with negative values, that would be wrong. I think that's a Python problem, because `-0` is probably converted to 0 before it makes it to NumPy, so there's not much we can do about that. But:

```python3
stats.hmean([0])  # 0.0
stats.hmean([np.NZERO])  # -0.0
stats.hmean([0, np.NZERO])  # nan, as it should be... but there's no warning. Oh well.
```